include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+//        Antony Polukhin, Ed Brey, Mark Rodgers, 
+//        Peter Dimov, and James Curran
+// when:  July 2001, April 2013 - 2019
+
+#include <algorithm>
+
+#include <boost/config.hpp>
+#include <boost/aligned_storage.hpp>
+#include <boost/type_index.hpp>
+#include <boost/type_traits/remove_reference.hpp>
+#include <boost/type_traits/decay.hpp>
+#include <boost/type_traits/remove_cv.hpp>
+#include <boost/type_traits/add_reference.hpp>
+#include <boost/type_traits/is_reference.hpp>
+#include <boost/type_traits/is_const.hpp>
+#include <boost/type_traits/is_nothrow_move_constructible.hpp>
+#include <boost/throw_exception.hpp>
+#include <boost/static_assert.hpp>
+#include <boost/utility/enable_if.hpp>
+#include <boost/core/addressof.hpp>
+#include <boost/type_traits/is_same.hpp>
+#include <boost/type_traits/is_const.hpp>
+#include <boost/type_traits/conditional.hpp>
+
+namespace boost
+{
+    template<std::size_t OptimizeForSize = sizeof(void*), std::size_t OptimizeForAlignment = boost::alignment_of<void*>::value>
+    class basic_any
+    {
+        BOOST_STATIC_ASSERT_MSG(OptimizeForSize > 0 && OptimizeForAlignment > 0, "Size and Align shall be positive values");
+        BOOST_STATIC_ASSERT_MSG(OptimizeForSize >= OptimizeForAlignment, "Size shall non less than Align");
+        BOOST_STATIC_ASSERT_MSG((OptimizeForAlignment & (OptimizeForAlignment - 1)) == 0, "Align shall be a power of 2");
+        BOOST_STATIC_ASSERT_MSG(OptimizeForSize % OptimizeForAlignment == 0, "Size shall be multiple of alignment");
+    private:
+        enum operation
+        {
+            Destroy,
+            Move,
+            Copy,
+            AnyCast,
+            UnsafeCast,
+            Typeinfo,
+        };
+
+        template <typename ValueType>
+        static void* small_manager(operation op, basic_any& left, const basic_any* right, const boost::typeindex::type_info* info)
+        {
+            switch (op)
+            {
+                case Destroy:
+                    reinterpret_cast<ValueType*>(&left.content.small_value)->~ValueType();
+                    break;
+                case Move:
+                    new (&left.content.small_value) ValueType(std::move(*reinterpret_cast<ValueType*>(&const_cast<basic_any*>(right)->content.small_value)));
+                    left.man = right->man;
+                    reinterpret_cast<ValueType const*>(&right->content.small_value)->~ValueType();
+                    const_cast<basic_any*>(right)->man = 0;
+                    break;
+                case Copy:
+                    new (&left.content.small_value) ValueType(*reinterpret_cast<const ValueType*>(&right->content.small_value));
+                    left.man = right->man;
+                    break;
+                case AnyCast:
+                    return boost::typeindex::type_id<ValueType>() == *info ?
+                            reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(&left.content.small_value) : 0;
```

Comment:
BOOST_DEDUCED_TYPENAME is meant for ancient MSVC, that had problems with templates. I do not think that we should support a two decades old compiler in this new implementation.

So just use `typename`

Reply:
done